uxarray/helpers.py

```diff
@@ -82,10 +84,149 @@ def determine_file_type(filepath):
-            os._exit(0)
+            raise ValueError(msg)
 
     return mesh_filetype
+
+
+def spherical_to_cartesian_unit(lat, lon, r=6371):
+    lat, lon = np.deg2rad(lat), np.deg2rad(lon)
+    x = r * np.cos(lat) * np.cos(lon)  # x coordinate
+    y = r * np.cos(lat) * np.sin(lon)  # y coordinate
+    z = r * np.sin(lat)  # z coordinate
+
+    coord = np.array([x, y, z])
+    # make it coord on a sphere with unit radius
+    unit_coord = coord / np.linalg.norm(coord)
+
+    return unit_coord
+
+
+def spherical_to_cartesian_unit(node, r=6371):
+    """Converts spherical (lat/lon) coordinates to cartesian (x,y,z).
+
+    Final output is cartesian coordinates on a sphere of unit radius
+
+    Parameters:
+    -----------
+
+    node: a list consisting of lat and lon
+
+    Returns: numpy array
+        Cartesian coordinates of length 3
+    """
+    lon = node[0]
+    lat = node[1]
+    lat, lon = np.deg2rad(lat), np.deg2rad(lon)
+    x = r * np.cos(lat) * np.cos(lon)  # x coordinate
+    y = r * np.cos(lat) * np.sin(lon)  # y coordinate
+    z = r * np.sin(lat)  # z coordinate
+
+    coord = np.array([x, y, z])
+    # make it coord on a sphere with unit radius
+    unit_coord = coord / np.linalg.norm(coord)
+
+    return unit_coord
+
+
+# Calculate the area of all faces.
+def calculate_face_area(x, y, z, coords_type="spherical"):
+    """Calculate area of a face on sphere.
+
+    Parameters
+    ----------
+
+    x : list, required
+        x-coordinate of all the nodes forming the face
+
+    y : list, required
+        y-coordinate of all the nodes forming the face
+
+    z : list, required
+        z-coordinate of all the nodes forming the face
+
+    coords_type : str, optional
+        coordinate type, default is spherical, can be cartesian also.
+    """
+    area = 0.0  # set area to 0
+    order = 6
+    dG, dW = get_gauss_quadratureDG(order)
+
+    num_nodes = len(x)
+
+    # num triangles is two less than the total number of nodes
+    num_triangles = num_nodes - 2
+    # Using tempestremap GridElements: https://github.com/ClimateGlobalChange/tempestremap/blob/master/src/GridElements.cpp
+    # loop thru all triangles
+    for j in range(0, num_triangles):
+        node1 = [x[0], y[0], z[0]]
+        node2 = [x[j + 1], y[j + 1], z[j + 1]]
+        node3 = [x[j + 2], y[j + 2], z[j + 2]]
+        if (coords_type == "spherical"):
+            node1 = spherical_to_cartesian_unit(node1)
+            node2 = spherical_to_cartesian_unit(node2)
+            node3 = spherical_to_cartesian_unit(node3)
+        for p in range(len(dW)):
+            for q in range(len(dW)):
+                dA = dG[p]
+                dB = dG[q]
+                jacobian = calculate_spherical_triangle_jacobian(
+                    node1, node2, node3, dA, dB)
+                area += dW[p] * dW[q] * jacobian
+    return area
+
+
+def calculate_spherical_triangle_jacobian(node1, node2, node3, dA, dB):
+    """Helper function for calculating face area."""
```

Comment:
For consistency with the rest of the file, I think this should have a full docstring